internal/storage/fs/cache.go

```diff
@@ -182,13 +184,27 @@ func (c *SnapshotCache[K]) References() []string {
 func (c *SnapshotCache[K]) evict(ref string, k K) {
 	logger := c.logger.With(zap.String("reference", ref))
 	logger.Debug("reference evicted")
-	for _, key := range append(maps.Values(c.fixed), c.extra.Values()...) {
-		if key == k {
-			return
-		}
+	if slices.Contains(append(maps.Values(c.fixed), c.extra.Values()...), k) {
+		return
 	}
 
 	delete(c.store, k)
 
 	logger.Debug("snapshot evicted", zap.String("key", fmt.Sprintf("%v", k)))
 }
+
+// Delete removes a reference from the snapshot cache.
+func (c *SnapshotCache[K]) Delete(ref string) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	if _, ok := c.fixed[ref]; ok {
+		delete(c.fixed, ref)
+		// No need to evict here, as fixed entries are not in LRU
+		return
+	}
```

Comment:
Fixed should always be present as its configured via Flipt configuration.
If it is deleted in the upstream we have a problem (e.g. `main` is put in fixed).